src/NLog/LayoutRenderers/ExceptionLayoutRenderer.cs

```diff
@@ -285,11 +296,29 @@ private void AppendException(Exception currentException, IEnumerable<ExceptionRe
                 int beforeRenderLength = builder.Length;
                 var currentRenderFunction = _renderingfunctions[renderingFormat];
                 currentRenderFunction(builder, currentException);
+
                 if (builder.Length == beforeRenderLength && builder.Length != orgLength)
                 {
                     builder.Length = orgLength;
                 }
             }
+
+            if (_customRenderingfunctions.TryGetValue(currentException.GetType(), out var renderer))
```

Comment:
Is this isn't a issue for subclasses?

Reply:
@304NotModified Do you mean that if there is no custom renderer for this type it should use a renderer for base type?

Reply:
If you have the classes like this,

    Exception <- MyExceptionBase <- MyException

And registering for MyExceptionBase, it won't work for MyException?

Reply:
You could use `IsAssignableFrom` for this, unfortunately it's not working a dictionary then. (or you should write a fancy comparer?)

Reply:
I mean IsInstanceOfType

Reply:
@304NotModified I just loop thought the base types if there is no renderer for current exception type